[Benchmark] Support M4Bench

Thorin215 opened the pull request:
Support [M4Bench](https://github.com/eaglelab-zju/M4Bench) [IJCAI 2025]

Thorin215 commented:
Hi @kennymckormick @niboshi @mfarre @gnobitab
I'm writing to beg for some help with a persistent CI failure on my PR https://github.com/open-compass/VLMEvalKit/pull/1163 . The vlm_test job is consistently failing. 

I've reviewed my changes but can't spot an obvious reason for a major performance issue. I am certain that the modifications I made will not affect files such as MMBench and OCRBench in the test. Could you please take a look when you have a moment or offer any suggestions on what might be causing the runner to terminate the process?

Any help would be greatly appreciated.

kennymckormick commented:
Hi, @Thorin215 

The CI problems might be due to that you develop based on an earlier commit, after which we updated the CI. So currently you can ignore it. 

However, I noticed several problems in this PR, which makes this benchmark not usable by users:
1. In m4bench.py, you used several interfaces we have not defined, like parse_choice or has_image, which leads to errors
2. In your tsv file, you only save the urls to the images, which is not a recommended practice and is not correctly handled by your `build_prompt`. We recommend you to save image as base64 format in the tsv file. 

Please fix those issues and make sure you can successfully run the evaluation under VLMEvalKit.

Thorin215 commented:
Hi @kennymckormick @niboshi @mfarre @gnobitab
Apologies for the inconvenience caused by the errors in my code. I have rectified the issues and verified the functionality locally using QwenVLPlus. Thank you for your time and consideration.

Thorin215 commented:
Hi @kennymckormick Thank you for your assistance. I have noticed that only one test remains unpassed, with the error pertaining to code format verification. However, I ensured that the pre-commit format check was executed prior to committing the code.  Could you provide me with some recommendations regarding this matter?

Thorin215 commented:
Hi @kennymckormick  Thank you very much for your detailed and prompt response. I have duly addressed your suggestions and truly appreciate your assistance.
